src/org/sosy_lab/common/collect/OrderStatisticSet.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  SoSy-Lab Common is a library of useful utilities.
+ *  This file is part of SoSy-Lab Common.
+ *
+ *  Copyright (C) 2007-2015  Dirk Beyer
+ *  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.sosy_lab.common.collect;
+
+import java.util.SortedSet;
+
+public interface OrderStatisticSet<T> extends SortedSet<T> {
+
+  T get(int pIndexOf);
+
+  T remove(int pIndexOf);
```

Comment:
Having `remove(int)` is problematic for `OrderStatisticSet<Integer>`, because calling `remove` with an `Integer` could call both methods. Better rename these two methods.

Reply:
Good point, I renamed them to {get,remove}ByRank , so that they are still close to the get/remove methods and easily findable.